src/broker/runlevel.c

```diff
@@ -196,13 +196,13 @@ static int runlevel_start_subprocess (struct runlevel *r, int level)
         if (flux_subprocess_aux_set (p, "runlevel", r, NULL) < 0)
             goto error;
 
-        monotime (&r->rc[level].start);
```

Comment:
this removal seems out of place in this commit?

Reply:
I think it's _probably_ OK.  The monotime call is moved to after the conditional so the level start time is initialized unconditionally, not just when there is a subprocess.

I included it with the commit that allows rc2 to not have a subprocess because it's noticeable if a garbage time value is printed when rc2 exits.  The fix applies equally to the other run levels, but when there is no subprocess, they are transitioned through instantaneously so time is not interesting, and I think we only do that in test.
```diff
diff --git a/src/broker/runlevel.c b/src/broker/runlevel.c
index c75a8c913..3124e2df6 100644
--- a/src/broker/runlevel.c
+++ b/src/broker/runlevel.c
@@ -196,13 +196,13 @@ static int runlevel_start_subprocess (struct runlevel *r, int level)
         if (flux_subprocess_aux_set (p, "runlevel", r, NULL) < 0)
             goto error;

-        monotime (&r->rc[level].start);
-
         r->rc[level].p = p;
-    } else {
+    }
+    else if (level == 1 || level == 3) {
         if (r->cb)
             r->cb (r, r->level, 0, 0., "Not configured", r->cb_arg);
     }
+    monotime (&r->rc[level].start);
     return 0;

 error:
```
Let me know what you think.  I could split it out easy enough.

Reply:
Er, the fix only applies to the rc2 with no subprocess case because when rc1 or rc3 have no subprocess, the callback is made directly with the elapsed time hardwired to zero.